front/src/modules/simulationResult/components/ChartHelpers/defineChart.ts

```diff
@@ -77,6 +80,33 @@ const defineChart = (
   const yAxisGrid = svg.append('g').attr('class', 'grid').call(gridY(y, width));
   const originalScaleY = y; // We need to keep a ref on that to not double translation
 
+  // Add Y axis on the right
+  let y2: SimulationD3Scale;
+  let y2Axis: d3.Selection<SVGGElement, unknown, null, undefined>;
+  let y2AxisGrid: d3.Selection<SVGGElement, unknown, null, undefined>;
+  let originalScaleY2: SimulationD3Scale;
+  let rightAxis;
+
+  if (defineY2) {
+    y2 = defineY2.range([height, 0]) as SimulationD3Scale;
+    const axisRightY = isSpaceSlopesCurves(keyValues) ? d3.axisRight(y2) : undefined;
```

Comment:
You can move the condition ` isSpaceSlopesCurves(keyValues) ` line 90 and remove the 2nd if loop
-> `if (defineY2 && isSpaceSlopesCurves(keyValues) ) {...}`
```suggestion
    const axisRightY = d3.axisRight(y2);
```